pkg/attempt.go

```diff
@@ -34,22 +34,23 @@ type Attempt struct {
 	NextRetryAfter time.Time `json:"nextRetryAfter,omitempty" bson:"nextRetryAfter,omitempty"`
 }
 
-func MakeAttempt(ctx context.Context, httpClient *http.Client, schedule []time.Duration, webhookID string, attemptNb int, cfg Config, data []byte) (Attempt, error) {
-	req, err := http.NewRequestWithContext(ctx, http.MethodPost, cfg.Endpoint, bytes.NewBuffer(data))
+func MakeAttempt(ctx context.Context, httpClient *http.Client, schedule []time.Duration, id string, attemptNb int, cfg Config, payload []byte) (Attempt, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, cfg.Endpoint, bytes.NewBuffer(payload))
 	if err != nil {
 		return Attempt{}, errors.Wrap(err, "http.NewRequestWithContext")
 	}
 
-	date := time.Now().UTC()
-	signature, err := security.Sign(webhookID, date, cfg.Secret, data)
+	ts := time.Now().UTC()
+	timestamp := ts.Unix()
+	signature, err := security.Sign(id, timestamp, cfg.Secret, payload)
 	if err != nil {
 		return Attempt{}, errors.Wrap(err, "security.Sign")
 	}
 
 	req.Header.Set("content-type", "application/json")
 	req.Header.Set("user-agent", "formance-webhooks/v0")
-	req.Header.Set("formance-webhook-id", webhookID)
-	req.Header.Set("formance-webhook-timestamp", fmt.Sprintf("%d", date.Unix()))
+	req.Header.Set("formance-webhook-id", id)
+	req.Header.Set("formance-webhook-timestamp", fmt.Sprintf("%d", timestamp))
```

Comment:
Maybe add a header indicating it is a test? It is common for this kind of feature, and it allow the targeted webhooks to take appropriate decision if needed.

Reply:
Good point! Do you have idea of the format?